### DISCUSSION TITLE
How aboute join the effort.

### DISCUSSION THREAD

**Discussion by chiddekel** [unoplatform/uno#7397] [Q&A]
How about joining the efforts with software developer of [AvaloniaUI](https://github.com/AvaloniaUI)

**Comment by robloo** [comment]
> As for the unrelated topic of Uno sitting on top of Avalonia, it doesn't make sense to us. First, we already have a Skia backend if/when needed. Second, we took an explicit direction of aligning with Microsoft's ecosystem so third party components can easily be reused.

https://github.com/unoplatform/uno/discussions/6097#discussioncomment-967738

Goals and architecture of the two projects are quite different.

**Comment by robloo** [comment]
I thought of some benefits in doing this. Avalonia now provides implementations for advanced text scenarios (Full Unicode, IME, TextFormatting, etc.) and a platform-independent input/pointer/touch API already similar to UWP. This would allow Uno to combine the GTK, WPF backends into one and also provide the same level of support for macOS -- that is a big simplification. This would essentially be a Skia backend though and have the same limitations with native control integration. Avalonia is quite stable for desktop these days and building out the WinUI API on top of it would be a winner. I don't think the projects would actually compete in that scenario.
